src/kbmod/masking.py

```diff
@@ -2,7 +2,42 @@
 """
 
 import kbmod.search as kb
+from kbmod.search import RawImage
 
+def mask_trajectory(trj, stack, r, bitmask = 1):
+    """ Will apply a circular mask of radius `r` around
+    the trajectory's predicted position in each image.
+
+    Attributes
+    ----------
+    trj : `Trajectory`
+        The trajectory along which to apply the mask.
+    stack : `ImageStack`
+        The stack of images to apply the mask to.
+    r : `float`
+        The radius of the circular mask to apply.
+
+    Returns
+    -------
+    stack : `ImageStack`
+        The stack after the masks have been applied.
+    """
+
+    dt = stack.get_single_image(1).get_obstime() - stack.get_single_image(0).get_obstime()
+    width = stack.get_single_image(0).get_width()
+    height = stack.get_single_image(0).get_height()
+
+    for i in range(stack.img_count()):
+        origin_of_mask = (trj.x + dt * trj.vx * i , trj.y + dt * trj.vy * i) 
+        trajectory_mask = RawImage(width, height)
+
+        for x in range(width):
+            for y in range(height):
+                if (x - origin_of_mask[0])**2 + (y - origin_of_mask[1])**2 <= r**2:
+                    trajectory_mask.set_pixel(y,x,bitmask)
```

Comment:
You are going to need to bounds check here. `set_pixel` will throw an error if you try to set a pixel that is off the image and trajectories might go off image.

Reply:
Won't `x` and `y` always be within the image here because of `y in range(height)` and `x in range(width)` where the `width` and `height` are the actual dimensions of the image?

Reply:
Yes. Sorry about that. You are correct.

I was thinking about a loop in the format of iterating over the dx and dy for the radius. I've added a comment about an alternate way to do this.

Reply:
Switched to this